DMCompiler/DM/Builders/DMExpressionBuilder.cs

```diff
@@ -338,25 +339,21 @@ private DMExpression BuildExpression(DMASTExpression expression, DreamPath? infe
                     BuildExpression(locateCoordinates.Y, inferredPath),
                     BuildExpression(locateCoordinates.Z, inferredPath));
                 break;
+            case DMASTAsType asType: {
+                var lhs = BuildExpression(asType.LHS, inferredPath);
+                var rhs = BuildExpression(asType.RHS, lhs.Path);
+
+                result = new AsType(asType.Location, lhs, rhs);
+                break;
+            }
             case DMASTIsSaved isSaved:
                 result = new IsSaved(isSaved.Location, BuildExpression(isSaved.Value, inferredPath));
                 break;
             case DMASTIsType isType: {
-                if (isType.RHS is DMASTIdentifier { Identifier: "__IMPLIED_TYPE__" }) {
```

Comment:
How is this getting used then? I don't see the logic moved

Reply:
The code was slightly redundant and made up for a broken inferred path. It's handled by [this](https://github.com/OpenDreamProject/OpenDream/blob/master/DMCompiler/DM/Builders/DMExpressionBuilder.cs#L571) now.